src/lib/HST/Algo.hs

```diff
@@ -1,16 +1,15 @@
 -- | This module contains the actual implementation of the pattern-matching
 --   compilation algorithm.
 
-module Algo
+module HST.Algo
```

Comment:
I'd suggest to rename the module -- at least -- to `Algorithm`. Perhaps there is a better name, maybe `Core`? Not sure about that, suggestions are welcome : ) @FreeProving/front-end

Reply:
Maybe `CoreAlgorithm`?

Reply:
Sure, best of both worlds ; )